### PR TITLE
enable non-{/32,/128} prefixes in cilium prefilter

### PR DESCRIPTION
**Pull request by borkmann**
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/9380)

### CODE REVIEW

**Comment by borkmann** [comment]
test-me-please

File: pkg/datapath/prefilter/prefilter.go [hunk 1]
```diff
@@ -277,13 +278,52 @@ func ProbePreFilter(device, mode string) error {
...
+func (p *probeKey) String() string             { return fmt.Sprintf("key=%d", p.Key) }
+func (p *probeKey) GetKeyPtr() unsafe.Pointer  { return unsafe.Pointer(p) }
+func (p *probeKey) NewValue() bpf.MapValue     { return &probeValue{} }
+func (p *probeKey) DeepCopyMapKey() bpf.MapKey { return &probeKey{p.Prefixlen, p.Key} }
+
+func (p *probeValue) String() string                 { return fmt.Sprintf("value=%d", p.Value) }
+func (p *probeValue) GetValuePtr() unsafe.Pointer    { return unsafe.Pointer(p) }
+func (p *probeValue) DeepCopyMapValue() bpf.MapValue { return &probeValue{p.Value} }
+
+func ProbePreFilterDyn() bool {
```
[the remark this answers is not in the document]
Comment (joestringer) [reply pkg/datapath/prefilter/prefilter.go:299]: It'd be nice to get some logging in here:
* If we decide this is not supported, info log that prefixes < full cidr are not supported in prefilter due to kernel version
* At least debug level logs for the error cases. This is not checking the specific kinds of failures at the moment (which is generally likely to work how we expect, but could catch false positives too).

**Review by joestringer — APPROVED** [review]
LGTM. We could put the new logic in a new file and get some more logging, but overall it looks like this PR does exactly what it says on the box. 

At some point it would be nice to migrate this functionality into bpftool.

**Comment by borkmann** [comment]
> LGTM. We could put the new logic in a new file and get some more logging, but overall it looks like this PR does exactly what it says on the box.

I have some follow-up changes for this tonight due to #9382. So I'll move the ProbePreFilterDyn() out and into its own 'probe' package where we can collect these things and later move to libbpf for the golang stuff. I'll place this into the other PR.

**Pull request merged**